src/db-manifest.js

```diff
@@ -1,13 +1,22 @@
 const path = require('path')
+const { DAGNode } = require('ipld-dag-pb')
 
 // Creates a DB manifest file and saves it in IPFS
-const createDBManifest = async (ipfs, name, type, accessControllerAddress) => {
+const createDBManifest = async (ipfs, name, type, accessControllerAddress, onlyHash) => {
   const manifest = {
     name: name,
     type: type,
     accessController: path.join('/ipfs', accessControllerAddress),
   }
-  const dag = await ipfs.object.put(Buffer.from(JSON.stringify(manifest)))
+  let dag
+  const manifestJSON = JSON.stringify(manifest)
+  if (onlyHash) {
+    dag = await new Promise(resolve => {
+      DAGNode.create(Buffer.from(manifestJSON), (err, n) => { resolve(n) })
```

Comment:
I'd like to understand what the return values are here? Does  `DAGNode.create(...)` return the same hash as `ipfs.object.put(...)`?

Reply:
We should handle the possible `err` in the callback here. Maybe throw it (which would be caught by the surrounding try/catch?

Reply:
>I'd like to understand what the return values are here? Does DAGNode.create(...) return the same hash as ipfs.object.put(...)?

@haadcode correct, `DAGNode.create(...)` returns the same hash as `ipfs.object.put(...)`. I'm not clear on the specifics of the method, but I followed the example [here](https://github.com/ipfs/js-ipfs/issues/1205#issuecomment-371134571) that @aphelionz was kind enough to point me to.

>We should handle the possible err in the callback here. Maybe throw it (which would be caught by the surrounding try/catch?

Done 👍